lona/routing.py

```diff
@@ -35,27 +35,26 @@ def __init__(self, raw_pattern, view, name='', interactive=True,
 
         # string or regex
         else:
-            if self.raw_pattern.endswith(OPTIONAL_TRAILING_SLASH_PATTERN):
+            raw_pattern = self.raw_pattern
+
+            if raw_pattern.endswith(OPTIONAL_TRAILING_SLASH_PATTERN):
                 self.optional_trailing_slash = True
 
-            groups = ABSTRACT_ROUTE_RE.findall(self.raw_pattern)
+                raw_pattern = \
+                    raw_pattern[:len(OPTIONAL_TRAILING_SLASH_PATTERN)*-1]
```

Comment:
`*-1` seems a bit odd here, why not `- len(OPTIONAL_TRAILING_SLASH_PATTERN)`

Reply:
Ack! Fixed